openhands/runtime/impl/docker/docker_runtime.py

```diff
@@ -242,27 +243,55 @@ def _init_container(self):
         environment.update(self.config.sandbox.runtime_startup_env_vars)
 
         self.log('debug', f'Workspace Base: {self.config.workspace_base}')
+        # Initialize volumes dictionary
+        volumes = {}
+        
+        # Add workspace mount if configured
         if (
             self.config.workspace_mount_path is not None
             and self.config.workspace_mount_path_in_sandbox is not None
         ):
             # e.g. result would be: {"/home/user/openhands/workspace": {'bind': "/workspace", 'mode': 'rw'}}
-            volumes = {
-                self.config.workspace_mount_path: {
-                    'bind': self.config.workspace_mount_path_in_sandbox,
-                    'mode': 'rw',
-                }
+            volumes[self.config.workspace_mount_path] = {
+                'bind': self.config.workspace_mount_path_in_sandbox,
+                'mode': 'rw',
             }
             logger.debug(f'Mount dir: {self.config.workspace_mount_path}')
         else:
             logger.debug(
                 'Mount dir is not set, will not mount the workspace directory to the container'
             )
+        
+        # Add Docker socket mount if enabled via environment variable or config
+        env_docker_socket_enabled = os.environ.get('SANDBOX_DOCKER_SOCKET_ENABLED', '').lower() == 'true'
+        docker_socket_enabled = env_docker_socket_enabled or self.config.sandbox.docker_socket_enabled
+        
+        # Get Docker socket path from environment variable or config
+        env_docker_socket_path = os.environ.get('SANDBOX_DOCKER_SOCKET_PATH', '')
+        docker_socket_path = env_docker_socket_path or self.config.sandbox.docker_socket_path
```

Comment:
These env vars automatically override `self.config.sandbox`, so there's no need to check both